plugins/chat/spec/system/chat_channel_spec.rb

```diff
@@ -426,4 +426,46 @@
 
     expect(page).to have_title("#🐕 Dogs - Chat - Discourse")
   end
+
+  context "when messages are sent with client timestamps" do
+    it "orders messages by client timestamp rather than server timestamp" do
+      first_message =
+        Chat::CreateMessage.call(
+          params: {
+            chat_channel_id: channel_1.id,
+            message: "I was created first but should appear second",
+            client_created_at: 30.seconds.ago.iso8601,
+          },
+          guardian: current_user.guardian,
+        ).message_instance
+
+      second_message =
+        Chat::CreateMessage.call(
+          params: {
+            chat_channel_id: channel_1.id,
+            message: "I was created second but should appear first",
+            client_created_at: 45.seconds.ago.iso8601,
+          },
+          guardian: current_user.guardian,
+        ).message_instance
+
+      chat_page.visit_channel(channel_1)
+
+      expect(page).to have_selector(
+        ".chat-message-container[data-id='#{first_message.id}'] .chat-message-text",
```

Comment:
You can use the message_by_id inside the PageObjects::Pages::ChatChannel for this